docs/reference/node/overview.md

```diff
@@ -150,11 +150,70 @@ globalThis.platformatic.messaging.handle('ping', () => 'pong')
 const timeoutId = setTimeout(() => console.log('done'), 10_000)
 
 // Optionally provide a close function
-export async function close() {
+export async function close () {
   clearTimeout(timeoutId)
 }
 ```
 
+## Handling Application Shutdown
+
+When your Node.js application needs to gracefully shut down, Platformatic provides several mechanisms to ensure resources are properly cleaned up:
+
+### `close` Function
+
+Export a `close` function from your application entry point to handle cleanup when the application shuts down:
+
+```js
+// Store references to resources that need cleanup
+let server
+let database
+
+export async function close () {
+  console.log('Cleaning up...')
+
+  // Clean up your resources
+  if (database) {
+    await database.close()
+  }
+  if (server) {
+    await server.close()
+  }
+
+  console.log('Application closed gracefully')
+}
+```
+
+### `close` Event Handler
+
+Alternatively, you can register a close event handler using the global Platformatic events:
+
+```js
+globalThis.platformatic.events.on('close', () => {
+  console.log('Received close event, cleaning up...')
+
+  // Perform your cleanup operations
+})
+```
+
+### Fastify Applications
+
+Fastify applications are handled automatically by Platformatic, but custom cleanup logic can still be implemented using the other mechanisms above if needed.
+
+### General behavior
+
+Platformatic Node handles closing the main application components:
+
+- **Applications with `create` function**: It will invoke the `close` method on the server returned by the function.
+- **Applications without `create` function**: It will invoke the `close` method on the first `node:http` server that listened on a TCP port.
+
+However, **additional resources must be manually closed** using the mechanisms described above, otherwise the application will hang during shutdown and eventually timeout.
+
+In applications launched via custom commands only the `close` event handler is available for cleanup and the `close` function is ignored.
+
+:::warn
+If your application is not using Fastify, doesn't export a `close` function and doesn't handle the `close` event, Platformatic will log a warning message suggesting you implement proper cleanup to avoid exit timeouts.
```

Comment:
```suggestion
If your application needs to clean up some shared states (connection pool, etc), you must export a `close` function or handle the `close` event. If you don't, Platformatic will log a warning message suggesting you implement proper cleanup to avoid exit timeouts. The exception is Fastify. 
```